sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1268,59 +1268,59 @@ ESIMD_NODEBUG ESIMD_INLINE
+  ESIMD_NODEBUG ESIMD_INLINE                                                   \
+      typename sycl::detail::enable_if_t<std::is_integral<T>::value cond,      \
+                                         simd<T1, N>>                          \
+          esimd_##op_hi(simd<T, N> src) {                                      \
+    return __esimd_##op_lo<T, N>(src.data());                                  \
+  }                                                                            \
+                                                                               \
+  template <typename T>                                                        \
+  ESIMD_NODEBUG ESIMD_INLINE                                                   \
+      typename sycl::detail::enable_if_t<std::is_integral<T>::value cond, T1>  \
+          esimd_##op_hi(T src) {                                               \
+    simd<T, 1> Src = src;                                                      \
+    simd<T1, 1> Result = esimd_##op_hi(Src);                                   \
+    return Result[0];                                                          \
+  }
 
+/// Count number of bits set in the source operand per element.
+/// @param src0 the source operand to count bits in.
+/// @return a vector of \c uint32_t, where each element is set to bit count of
+///     the corresponding element of the source operand.
+__ESIMD_DEFINE_BIT_OP(uint32_t, cbit, cbit, &&sizeof(T) <= 4)
+
+/// Find the per element number of the first bit set in the source operand
+/// starting from the least significant bit.
+/// @param src0 the source operand to count bits in.
+/// @return a vector of the same type as the source operand, where each element
+///     is set to the number first bit set in corresponding element of the
+///     source operand. \c 0xFFFFffff is returned for an element equal to \c 0.
 /// Find component-wise the first bit from LSB side
-template <int N>
-ESIMD_NODEBUG ESIMD_INLINE simd<unsigned, N> esimd_fbl(simd<unsigned, N> src) {
-  return __esimd_fbl(src.data());
-}
-
-template <typename T = void>
-ESIMD_NODEBUG ESIMD_INLINE unsigned esimd_fbl(unsigned src) {
-  simd<unsigned, 1> Src = src;
-  simd<unsigned, 1> Result = esimd_fbl(Src);
-  return Result[0];
-}
-
-/// Find component-wise the first bit from MSB side.
-template <int N>
-ESIMD_NODEBUG ESIMD_INLINE simd<int, N> esimd_fbh(simd<int, N> src) {
-  return __esimd_sfbh(src.data());
-}
-
-template <int N>
-ESIMD_NODEBUG ESIMD_INLINE simd<unsigned int, N>
-esimd_fbh(simd<unsigned int, N> src) {
-  return __esimd_ufbh(src.data());
-}
-
-template <typename T>
-ESIMD_NODEBUG ESIMD_INLINE typename sycl::detail::enable_if_t<
-    detail::is_dword_type<T>::value && detail::is_esimd_scalar<T>::value, T>
-esimd_fbh(T src) {
-  simd<T, 1> Src = src;
-  simd<T, 1> Result = esimd_fbh(Src);
-  return Result[0];
-}
-
-template <typename T = void> simd<uint, 4> esimd_rdtsc();
```

Comment:
Are there any codes that use `esimd_rdtsc` ? Maybe deprecate first?

Reply:
rdtsc has never been implemented, it is not lowered in LoweESIMD, any usage would yield:

atomic.cpp:68:22: error: SYCL kernel cannot call an undefined function without SYCL_EXTERNAL attribute
            auto x = esimd_rdtsc();

this is a dead code.